test/ranges.jl

```diff
@@ -1587,3 +1587,12 @@ end
     @test step(r) === 2
     @test collect(r) == ['a','c','e','g']
 end
+
+@testset "Char ranges are regular" begin
+    r = 'a':'z'
+    r2 = 'α':2:'ω'
+    r3 = StepRangeLen('a',3,10)
+    @test Base.RangeStepStyle(r) === Base.RangeStepRegular()
```

Comment:
Is there a test that failed before you set the step for char ranges to be regular? If so it would be good to test that as well. (It's generally better to test *outcomes* than *representations*, though I agree there's room for testing both. But representations tend to be more fragile with respect to internal changes within Julia, and what we ultimately care about is the outcome.)

Reply:
On master char ranges get the fast `findfirst` method, but they won't once this checks the trait. But there's no explicit error or wrong result, it's just slower. Agree this isn't much of a test though!